src/main/java/com/kaleido/fetch/service/FetchService.java

```diff
@@ -0,0 +1,353 @@
+package com.kaleido.fetch.service;
+
+import com.kaleido.fetch.domain.Component;
+import com.kaleido.kaptureclient.client.KaptureClient;
+import com.kaleido.kaptureclient.domain.Batch;
+import com.kaleido.kaptureclient.domain.BatchAlias;
+import com.kaleido.kaptureclient.domain.ChemicalConcept;
+import com.kaleido.kaptureclient.domain.Community;
+import com.kaleido.kaptureclient.domain.Media;
+import com.kaleido.kaptureclient.domain.Supplement;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+@Slf4j
+@Service
+public class FetchService {
+
+    private static final int URI_BATCH_SIZE = 50;
+    private static final String MICRO_LITER = "\u00B5L";
+    private static final String MICRO_GRAM = "\u00B5g";
+
+    private KaptureClient<Media> mediaKaptureClient;
+    private KaptureClient<Batch> batchKaptureClient;
+    private KaptureClient<Community> communityKaptureClient;
+    private KaptureClient<Supplement> supplementKaptureClient;
+
+    public FetchService(KaptureClient<Media> mediaKaptureClient, KaptureClient<Batch> batchKaptureClient, KaptureClient<Community> communityKaptureClient, KaptureClient<Supplement> supplementKaptureClient) {
+        this.mediaKaptureClient = mediaKaptureClient;
+        this.batchKaptureClient = batchKaptureClient;
+        this.communityKaptureClient = communityKaptureClient;
+        this.supplementKaptureClient = supplementKaptureClient;
+    }
+
+    public List<Component> findComponents(String searchTerm) {
+        log.debug("call to /components/search/{}", searchTerm);
```

Comment:
The log statement should probably move back to the resource class rather than the service class as it is related to the URL